interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala

```diff
@@ -0,0 +1,191 @@
+        // property is typed (#foo.bar), so we need to return filtered list of all methods and fields from previous spel node type
+        case p: PropertyOrFieldReference =>
+          suggestionsForPropertyOrFieldReference(nodeInPosition, p)
+        // suggestions for dictionary with indexer notation - #dict['Foo']
+        // 1. caret is inside string
+        // 2. parent node is Indexer - []
+        // 3. parent's prev node is dictionary
+        case s: StringLiteral =>
+          val y = for (
+            parent <- nodeInPosition.parent.map(_.node);
+            parentPrevNode <- parent.prevNode();
+            parentPrevNodeTyping <- parentPrevNode.typingResultWithContext.map(_.typingResult)
+          ) yield {
+            parent.spelNode match {
+              case _: Indexer => parentPrevNodeTyping match {
+                case td: TypedDict => dictQueryService.queryEntriesByLabel(td.dictId, s.getLiteralValue.getValue.toString)
+                  .map(_.map(list => list.map(e => ExpressionSuggestion(e.label, td, fromClass = false)))).getOrElse(successfulNil)
+                case _ => successfulNil
+              }
+              case _ => successfulNil
+            }
+          }
+          y.getOrElse(successfulNil)
+        case _ => successfulNil
+      }
+    }
+    suggestions.getOrElse(successfulNil).map(_.toList.sortBy(_.methodName))
+
+  }
+
+  private def insertDummyVariable(s: String, index: Int): String = {
+    val (start, end) = s.splitAt(index)
+    start + "x" + end
+  }
+}
+
+private class NuSpelNodeParser(val expressionConfig: ExpressionDefinition[_], val typeDefinitions: TypeDefinitionSet, val classLoader: ClassLoader) extends LazyLogging {
+  private val classResolutionStrategy = SupertypeClassResolutionStrategy.Union
+  private val commonSupertypeFinder = new CommonSupertypeFinder(classResolutionStrategy, expressionConfig.strictTypeChecking)
+  private val conversionService = determineConversionService(expressionConfig)
+
+  // fixme copy-pasted from SpelExpressionParser, move it to another class and reuse
+  private val functions = Map(
+    "today" -> classOf[LocalDate].getDeclaredMethod("now"),
+    "now" -> classOf[LocalDateTime].getDeclaredMethod("now"),
+    "distinct" -> classOf[CollectionUtils].getDeclaredMethod("distinct", classOf[java.util.Collection[_]]),
+    "sum" -> classOf[CollectionUtils].getDeclaredMethod("sum", classOf[java.util.Collection[_]])
+  )
+  private val propertyAccessors = internal.propertyAccessors.configured()
+
+  private val evaluationContextPreparer = new EvaluationContextPreparer(classLoader, expressionConfig.globalImports, propertyAccessors, conversionService, functions, expressionConfig.spelExpressionExcludeList)
+  private val parser = new org.springframework.expression.spel.standard.SpelExpressionParser()
+  private val typer = new Typer(
```

Comment:
Let's do some refactor, that will be easier to create the Typer to avoid code duplication. Let it be passed into `SpelExpressionParser.default` and into `SpelExpressionSuggester`/`NuSpelNodeParser`

Reply:
Fixed, now it should be a little bit easier to instantiate these objects.